### PR TITLE
PAYEE-741 - Added test for new redirect when cache is empty

### PR DESCRIPTION
**Pull request by brendanstraughan**
- [ ] Unit tests passing
- [ ] Coverage reached
- [ ] Acceptance tests passing
- [ ] Reviewed

### CODE REVIEW

File: test/controllers/IncomeControllerSpec.scala [hunk 1]
```diff
@@ -274,6 +275,31 @@ class IncomeControllerSpec extends BaseSpec with JourneyCacheConstants with I18n
...
+        val annualAccount = AnnualAccount(TaxYear(), Available, List(payment), Nil)
+        val employment = employmentWithAccounts(List(annualAccount))
+
+        val result =
+          testController.editRegularIncome(empId = employment.sequenceNumber)(
+            RequestBuilder.buildFakeRequestWithAuth("POST").withJsonBody(formData))
+
+        status(result) mustBe SEE_OTHER
+        redirectLocation(result) mustBe Some(
+          controllers.routes.IncomeSourceSummaryController.onPageLoad(employment.sequenceNumber).url)
```
Comment (brendanstraughan) [inline test/controllers/IncomeControllerSpec.scala:298]: could this be employerId, more readable

File: test/controllers/income/estimatedPay/update/IncomeUpdateIrregularHoursControllerSpec.scala [hunk 2]
```diff
@@ -286,7 +290,7 @@ class IncomeUpdateIrregularHoursControllerSpec extends BaseSpec with JourneyCach
     "respond with INTERNAL_SERVER_ERROR for failed request to cache" in {
 
       val result = ConfirmIncomeIrregularHoursHarness
-        .setup(failure = true)
+        .setup(failure = true, futureFailed = true)
```
Comment (brendanstraughan) [inline test/controllers/income/estimatedPay/update/IncomeUpdateIrregularHoursControllerSpec.scala:293]: which scenario are we testing here?

File: test/controllers/IncomeControllerSpec.scala (hunk 1 above)
Comment (PerrettJ4) [reply test/controllers/IncomeControllerSpec.scala:298]: yeah seems fair...

File: test/controllers/income/estimatedPay/update/IncomeUpdateIrregularHoursControllerSpec.scala (hunk 2 above)
Comment (PerrettJ4) [reply test/controllers/income/estimatedPay/update/IncomeUpdateIrregularHoursControllerSpec.scala]: I'm not sure tbh

File: app/views/incomes/HistoricIncomeCalculationView.scala.html [hunk 3]
```diff
@@ -149,6 +149,5 @@
     </div>
     </div>
     </div>
-</div>
```
Comment (PerrettJ4) [inline app/views/incomes/HistoricIncomeCalculationView.scala.html:152]: removed unmatched div here

**Review by vivekhmrc — APPROVED** [review]
👍

File: app/controllers/income/estimatedPay/update/IncomeUpdateCalculatorController.scala [hunk 4]
```diff
@@ -120,12 +124,11 @@ class IncomeUpdateCalculatorController @Inject()(
...
         .mandatoryJourneyValues(
           UpdateIncome_NameKey,
-          UpdateIncome_IdKey,
           s"$UpdateIncome_ConfirmedNewAmountKey-$empId",
           UpdateIncome_IncomeTypeKey)
         .getOrFail
-        .flatMap { mandatoryJourneyValues =>
-          val incomeName :: incomeId :: newAmount :: incomeType :: Nil = mandatoryJourneyValues.toList
+        .map { mandatoryJourneyValues =>
+          val incomeName :: newAmount :: incomeType = mandatoryJourneyValues.toList
```
Comment (sergi-hmrc) [inline app/controllers/income/estimatedPay/update/IncomeUpdateCalculatorController.scala:131]: if u remove that Nil then incomType is a List which means the if below will never satisfy the condition: 
`val vm = if (incomeType == TaiConstants.IncomeTypePension) {`

**Review by sergi-hmrc — APPROVED** [review]

**Pull request merged**